manifests/charts/gateway/templates/deployment.yaml

```diff
@@ -16,6 +16,10 @@ spec:
   selector:
     matchLabels:
       {{- include "gateway.selectorLabels" . | nindent 6 }}
+  strategy:
```

Comment:
So based on my interpretation of the docs here https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#rolling-update-deployment and what I've observed on my current istio ingress gateway deployment, you get a default strategy of 25% for each of the max surge and max unavailable options. So I think the only change here is that it's explicitly defined in the gateway chart now, but that there should be no actual change as people update their versions of istio to pull in this change. Let me know if your understanding is different though.

Reply:
will there an yaml change then cause an rollout of gateway deployment?

Reply:
I would anticipate it being a no-op since the resulting k8s deployment would be the same (it's just now explicitly defined in this chart), but I'll try to verify that when I get a chance today.

Reply:
I just pulled this change into my ingress gateway setup and when relying on the defaults there was no rollout, so I don't believe this change itself would trigger any rollouts of the gateway deployment as people upgrade.

Reply:
I think this field could be more flexible, such as:
```
strategy: {{- toYaml .Values.updateStrategy | nindent 4 }}
```
WDYT

Reply:
just made this change here if we want to reopen this PR https://github.com/Collin3/istio/commit/9c5030391d9677e4216c99c2bd9aaf2e5e9e23b8